### ISSUE TITLE
Focus Indicator fails contrast standards

### ISSUE THREAD

**Issue by naomi-lgbt** [freeCodeCamp/freeCodeCamp#53348]
### Describe the Issue

When moving keyboard focus to an element in our challenge list, the background-colour change combined with the focus indicator result in a contrast ratio less than 3:1.

**This issue appears to only occur in light mode.**

### Affected Page

Pick a superblock, any superblock.

### Steps to Reproduce

- Pick a superblock
- Expand a block section
- `TAB` to move focus to a challenge list item
- See that the background colour and border both change

### Expected behavior

Contrast needs to meet standards.

### Screenshots

| Example | Contrast |
| --- | --- |
| ![Image demonstrating the keyboard focus selection indicator in the C# superblock.](https://github.com/freeCodeCamp/freeCodeCamp/assets/63889819/47fa94ae-af1a-4337-bea6-4f4e83402ae0) | <img width="483" alt="WCAG contrast checker, showing that the background and focus indicator only have a contrast ratio of 2.56:1" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/63889819/0f593af4-320a-4d88-9648-6d367f2bec34"> |

### System

NaomiOS

### Additional context

Thanks to [Emma](https://twitter.com/EmmaDawsonDev/status/1750246088819585212) for catching this.

**Comment by bbsmooth** [comment]
We are OK here. The blue focus border around the link is above `3:1` with the background and that's all that's needed. The ratio between the blue border and the darker gray background isn't a concern. Think of it this way, if we didn't have the gray background change and instead, just the blue border, I think you would agree we would be fine. So we have an indicator that is sufficient no matter what happens inside of that border.

**Comment by bbsmooth** [comment]
Just wanted to add one more thought on this since I've spent some time thinking about it. As is typical with several WCAG SCs, the words used to explain the intent do not always mean what we normally think they mean. [SC 1.4.11](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html) starts off with:

> The visual presentation of the following have a contrast ratio of at least 3:1 against **adjacent** color(s):

To me, the word "adjacent" means touching or adjoining, so taken in that sense then yes, the gray is adjacent to the blue focus indicator and since there isn't enough contrast then it fails. But the understanding doc [clarifies](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html#figure-standard-text-adjacent-white) clarifies that this isn't exactly what is meant by "adjacent" here:

> For user interface components 'adjacent colors' means the colors adjacent to the component. For example, if an input has a white internal background, dark border, and white external background the 'adjacent color' to the component would be the white external background.

In other words, the adjacent colors to be compared for focus contrast aren't the colors in the focus indicator itself, but rather the color of the focus indicator compared to the background color of the component in its non-focused state. For the particular case we are discussing, since we aren't using additional color to define the link, the component is the link text itself and the component background is the white surrounding the text. The focus indicator is both the gray and blue that appear around the text. What we are comparing for contrast is the gray/blue focus indicator to the white background. And WCAG does not require that the entire focus indicator meet minimum contrast requirements. Technically, only one pixel in the focus indicator needs to meet the requirement to pass. Obviously that would make for a terrible focus indicator, but WCAG is a bare minimum, not necessarily a best practice.

**Comment by naomi-lgbt** [comment]
As always, thank you for your thorough and detailed writeup.

Given this information, I think it's fine to close this issue. :3

**Comment by EmmaDawsonDev** [comment]
Thanks @bbsmooth for the clarification. Gotta love that WCAG has it's own definition of adjacent that apparently doesn't mean adjacent 🙄

**Comment by bbsmooth** [comment]
Nothing will top "color" not actually meaning "color".
